src/test/java/org/opensearch/search/asynchronous/integTests/AsynchronousSearchStatsIT.java

```diff
@@ -166,11 +166,15 @@ public void testStatsAcrossNodes() throws InterruptedException, ExecutionExcepti
             TestThreadPool finalThreadPool = threadPool;
             threads.forEach(t -> finalThreadPool.generic().execute(t));
             latch.await();
-            AsynchronousSearchStatsResponse statsResponse = client().execute(AsynchronousSearchStatsAction.INSTANCE,
-                    new AsynchronousSearchStatsRequest()).get();
+
             AtomicLong actualNumSuccesses = new AtomicLong();
             AtomicLong actualNumFailures = new AtomicLong();
             AtomicLong actualNumPersisted = new AtomicLong();
+
+            AsynchronousSearchStatsResponse statsResponse = client().execute(AsynchronousSearchStatsAction.INSTANCE,
+                    new AsynchronousSearchStatsRequest()).get();
+            waitForAsyncSearchTasksToComplete();
```

Comment:
The previous line fetched the stats. Why would adding a wait here change the result of this test?

Reply:
Thanks, you're right that line is misplaced. Updated.